### PR TITLE
[CELEBORN-1331] Remove third-party dependencies in shaded clients' pom

### PR DESCRIPTION
**Pull request by kerwin-zk**
### What changes were proposed in this pull request?
celeborn-client-spark-3-shaded_2.12 should not expose the shipped transitive deps

### Why are the changes needed?
maven dependency

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
CI

### CODE REVIEW

**Comment by kerwin-zk** [comment]
@pan3793 cc

**Comment by kerwin-zk** [comment]
@pan3793 The Columnar Shuffle interface of Spark-3.5 has changed. I will support Spark-3.5 with  columnar shuffle. Once supported, the CI here can pass.

**Comment by pan3793** [comment]
I'm afraid such a change won't address the issue, 0.4.0 was released via SBT, I think the real issue comes from SBT build script

**Comment by kerwin-zk** [comment]
@pan3793 @cfmcgrady cc

**Comment by cfmcgrady** [comment]
After discussing with @pan3793 

1. It appears that the issue is not limited to the `celeborn-client-spark-3-shaded_2.12` alone. It would be advisable to address this issue for all shaded clients(spark-2/flink/mr).
2. Additionally, we should consider excluding all third-party dependencies and not just those related to the Celeborn client.
3. It would be helpful to add a comment explaining our approach to this matter.

Thank you.

**Comment by SteNicholas** [comment]
@kerwin-zk, any update?

**Comment by kerwin-zk** [comment]
> @kerwin-zk, any update?

Working on it.

File: project/CelebornBuild.scala [hunk 1]
```diff
@@ -389,6 +391,17 @@ object Utils {
     require(ALL_SCALA_VERSIONS.contains(v), s"found not allow scala version: $v")
     v
   }
+
+  def removeDependenciesTransformer: xml.Node => xml.Node = { node =>
```
Comment (cfmcgrady) [inline project/CelebornBuild.scala:395]: add a comment to explain why we are excluding all dependencies.

File: project/CelebornBuild.scala [hunk 2]
```diff
@@ -836,13 +854,11 @@ trait SparkClientProjects {
           case _ => MergeStrategy.first
         },
 
-        Compile / packageBin := assembly.value
+        Compile / packageBin := assembly.value,
+        publishMavenStyle := true,
```
[the remark this answers is not in the document]
Comment (kerwin-zk) [reply project/CelebornBuild.scala]: @cfmcgrady done.

File: project/CelebornBuild.scala (hunk 1 above)
Comment (kerwin-zk) [reply project/CelebornBuild.scala:399]: @cfmcgrady done

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main(v0.5.0)

**Comment by pan3793** [comment]
Merged to main/0.4

**Pull request closed without merging**